reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/SwitchIfEmpty.kt

```diff
@@ -0,0 +1,34 @@
+package com.badoo.reaktive.observable
+
+import com.badoo.reaktive.disposable.Disposable
+import com.badoo.reaktive.disposable.DisposableWrapper
+import com.badoo.reaktive.utils.atomicreference.AtomicReference
+
+fun <T> Observable<T>.switchIfEmpty(otherObservable: Observable<T>): Observable<T> =
+    observableUnsafe { observer ->
+        val disposableWrapper = DisposableWrapper()
+        observer.onSubscribe(disposableWrapper)
+
+        subscribeSafe(
+            object : ObservableObserver<T>, ObservableCallbacks<T> by observer {
+                private val isEmpty = AtomicReference(true)
+
+                override fun onSubscribe(disposable: Disposable) {
+                    disposableWrapper.set(disposable)
+                }
+
+                override fun onNext(value: T) {
+                    isEmpty.value = false
+                    observer.onNext(value)
+                }
+
+                override fun onComplete() {
+                    if (isEmpty.value) {
+                        otherObservable.subscribe(this)
```

Comment:
1. Please use 'subscribeSafe'
2. You are subscribing with the same observer, if second observable will be empty as well, then you will subscribe to it again. You should create a new observer and delegate only 'Observer' and 'ErrorCallback' to 'this'. Please cover this case with tests.